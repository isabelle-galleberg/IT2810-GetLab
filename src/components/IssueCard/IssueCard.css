.issueCard {
  background-color: #f2f1f0;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  border: 1px solid #3a73c5;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  gap: 5px;
  padding: 5px;
  height: 100%;
  width: clamp(200px, calc(75vw), 400px);
}

.issueTitle {
  font-weight: bold;
}

.label {
  background-color: #3a73c5;
  border-radius: 10px;
  color: white;
  padding: 3px;
  white-space: nowrap;
}

.labels {
  display: flex;
  flex-direction: row;
  gap: 5px;
  overflow: auto;
}

/* hide scrollbar for Chrome, Safari and Opera */
.labels::-webkit-scrollbar {
  display: none;
}

/* hide scrollbar for IE, Edge and Firefox */
.labels {
  -ms-overflow-style: none;
  /* IE and Edge */
  scrollbar-width: none;
  /* Firefox */
}

.issueInfo {
  display: flex;
  flex-direction: row;
  gap: 5px;
}

.createdAt {
  color: grey;
}

@media screen and (min-width: 500px) {
  .issueTitle,
  .issueNumber,
  .createdAt,
  .author {
    font-size: 12px;
  }
  .label {
    font-size: 10px;
  }
}

@media screen and (max-width: 500px) {
  .issueTitle,
  .issueNumber,
  .createdAt,
  .author {
    font-size: 10px;
  }
  .label {
    font-size: 8px;
  }
}
