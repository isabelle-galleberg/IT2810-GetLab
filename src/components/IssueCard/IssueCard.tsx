import "./IssueCard.css";
import { IssueCardProps } from "../../types/propTypes";

export default function IssueCard({
  title,
  labels,
  issueNumber,
  createdAt,
  author,
}: IssueCardProps) {
  return (
    <div className="issueCard">
      <div className="issueTitle">{title}</div>
      <div className="labels">
        {labels.map((label: string) => {
          return (
            <div className="label" key={label}>
              {label}
            </div>
          );
        })}
      </div>
      <div className="issueInfo">
        <div className="issueNumber">{`#${issueNumber}`}</div>
        <div className="createdAt">{`created ${createdAt} by`}</div>
        <div className="author">{author}</div>
      </div>
    </div>
  );
}
