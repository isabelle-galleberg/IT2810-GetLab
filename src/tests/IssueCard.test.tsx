import renderer from "react-test-renderer";
import IssueCard from "../components/IssueCard/IssueCard";

describe("<IssueCard />", () => {
  it("should render an issue card with given values", () => {
    const elem = renderer
      .create(
        <IssueCard
          title="Make it work"
          labels={["priority: high"]}
          issueNumber={1}
          createdAt="10.08.2022"
          author="me"
        />
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
