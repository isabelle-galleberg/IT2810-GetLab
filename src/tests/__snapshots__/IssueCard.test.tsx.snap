// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueCard /> should render an issue card with given values 1`] = `
<div
  className="issueCard"
>
  <div
    className="issueTitle"
  >
    Make it work
  </div>
  <div
    className="labels"
  >
    <div
      className="label"
    >
      priority: high
    </div>
  </div>
  <div
    className="issueInfo"
  >
    <div
      className="issueNumber"
    >
      #1
    </div>
    <div
      className="createdAt"
    >
      created 10.08.2022 by
    </div>
    <div
      className="author"
    >
      me
    </div>
  </div>
</div>
`;
